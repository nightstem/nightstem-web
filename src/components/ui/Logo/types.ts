import type {
  LOGO_ANIMATIONS,
  LOGO_MODES,
  LOGO_SIZES,
} from '@/components/ui/Logo/constants';

export type LogoMode = (typeof LOGO_MODES)[keyof typeof LOGO_MODES];

export type LogoSize = (typeof LOGO_SIZES)[keyof typeof LOGO_SIZES];

export type LogoAnimation =
  (typeof LOGO_ANIMATIONS)[keyof typeof LOGO_ANIMATIONS];

export type LogoSizeConfig = {
  textClass: string;
  logoClass: string;
  gapClass: string;
};
