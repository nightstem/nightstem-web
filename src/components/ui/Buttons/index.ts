export * from '@/components/ui/Buttons/Button';
export * from '@/components/ui/Buttons/LinkButton';
