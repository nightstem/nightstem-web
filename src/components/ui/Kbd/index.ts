export * from './Kbd';
export { default } from './Kbd';
