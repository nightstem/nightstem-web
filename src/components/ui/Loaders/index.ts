export * from '@/components/ui/Loaders/PageLoader';
