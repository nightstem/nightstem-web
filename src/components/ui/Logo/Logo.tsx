import cx from 'classnames';

import LogoMark from '@/assets/svgs/LogoMark';

import type {
  LogoMode,
  LogoSize,
  LogoAnimation,
} from '@/components/ui/Logo/types';

import {
  LOGO_MODES,
  LOGO_SIZES,
  LOGO_SIZE_CONFIG,
  LOGO_ANIMATIONS,
  LOGO_ANIMATION_CONFIG,
} from '@/components/ui/Logo/constants';

export type LogoProps = {
  mode?: LogoMode;
  size?: LogoSize;
  animation?: LogoAnimation;
};

export const Logo = ({
  mode = LOGO_MODES.ICON,
  size = LOGO_SIZES.BODY,
  animation = LOGO_ANIMATIONS.NONE,
}: LogoProps) => {
  const appName = 'Nightstem';
  const sizeConfig = LOGO_SIZE_CONFIG[size];
  const animationClass = LOGO_ANIMATION_CONFIG[animation];

  const logo = (
    <LogoMark
      title={`${appName} logo`}
      decorative={mode !== LOGO_MODES.ICON}
      className={cx(
        sizeConfig.logoClass,
        animationClass,
        'motion-reduce:animate-none',
        'shrink-0',
      )}
    />
  );

  switch (mode) {
    case LOGO_MODES.HORIZONTAL: {
      return (
        <div className={cx('inline-flex items-center', sizeConfig.gapClass)}>
          {logo}
          <span className={sizeConfig.textClass}>{appName}</span>
        </div>
      );
    }

    case LOGO_MODES.ICON:
    default: {
      return <div className="inline-flex">{logo}</div>;
    }
  }
};

Logo.displayName = 'Logo';
export default Logo;
