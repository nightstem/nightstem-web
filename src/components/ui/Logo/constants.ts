export const LOGO_MODES = {
  ICON: 'icon',
  HORIZONTAL: 'horizontal',
  FULL: 'full',
} as const;

export const LOGO_SIZES = {
  DISPLAY: 'display',
  XXL: 'xxl',
  XL: 'xl',
  LG: 'lg',
  MD: 'md',
  SM: 'sm',
  BODY: 'body',
  CAPTION: 'caption',
  HINT: 'hint',
} as const;

export const LOGO_ANIMATIONS = {
  NONE: 'none',
  PULSE: 'pulse',
} as const;

export const LOGO_ANIMATION_CONFIG = {
  [LOGO_ANIMATIONS.NONE]: '',
  [LOGO_ANIMATIONS.PULSE]: 'animate-pulse',
} as const;

export const LOGO_SIZE_CONFIG = {
  [LOGO_SIZES.DISPLAY]: {
    textClass: 'heading-display', // 5.61rem
    logoClass: 'h-[4.488rem] w-[4.488rem]', // 5.61 * 0.8
    gapClass: 'gap-4',
  },
  [LOGO_SIZES.XXL]: {
    textClass: 'heading-xxl', // 4.209rem
    logoClass: 'h-[3.367rem] w-[3.367rem]', // 4.209 * 0.8
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.XL]: {
    textClass: 'heading-xl', // 3.157rem
    logoClass: 'h-[2.526rem] w-[2.526rem]', // 3.157 * 0.8
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.LG]: {
    textClass: 'heading-lg', // 2.369rem
    logoClass: 'h-[1.895rem] w-[1.895rem]', // 2.369 * 0.8
    gapClass: 'gap-3',
  },
  [LOGO_SIZES.MD]: {
    textClass: 'heading-md', // 1.777rem
    logoClass: 'h-[1.422rem] w-[1.422rem]', // 1.777 * 0.8
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.SM]: {
    textClass: 'heading-sm', // 1.333rem
    logoClass: 'h-[1.066rem] w-[1.066rem]', // 1.333 * 0.8
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.BODY]: {
    textClass: 'text-body', // 1rem
    logoClass: 'h-[0.8rem] w-[0.8rem]', // 1 * 0.8
    gapClass: 'gap-2',
  },
  [LOGO_SIZES.CAPTION]: {
    textClass: 'text-caption', // 0.75rem
    logoClass: 'h-[0.6rem] w-[0.6rem]', // 0.75 * 0.8
    gapClass: 'gap-1',
  },
  [LOGO_SIZES.HINT]: {
    textClass: 'text-hint', // 0.563rem
    logoClass: 'h-[0.45rem] w-[0.45rem]', // 0.563 * 0.8
    gapClass: 'gap-1',
  },
} as const;
