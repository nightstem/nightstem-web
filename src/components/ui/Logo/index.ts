export * from '@/components/ui/Logo/Logo';
export { default } from '@/components/ui/Logo/Logo';
export type {
  LogoMode,
  LogoSize,
  LogoAnimation,
} from '@/components/ui/Logo/types';
export {
  LOGO_MODES,
  LOGO_SIZES,
  LOGO_ANIMATIONS,
} from '@/components/ui/Logo/constants';
