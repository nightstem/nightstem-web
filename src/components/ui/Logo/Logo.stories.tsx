import type { Meta, StoryObj } from '@storybook/nextjs-vite';

import {
  LOGO_MODES,
  LOGO_SIZES,
  LOGO_ANIMATIONS,
} from '@/components/ui/Logo/constants';
import { Logo, type LogoProps } from '@/components/ui/Logo';

const meta = {
  component: Logo,
  tags: ['autodocs'],
  argTypes: {
    mode: {
      control: { type: 'select' },
      options: Object.values(LOGO_MODES),
    },
    size: {
      control: { type: 'select' },
      options: Object.values(LOGO_SIZES),
    },
    animation: {
      control: { type: 'select' },
      options: Object.values(LOGO_ANIMATIONS),
    },
  },
} satisfies Meta<LogoProps>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Icon: Story = {
  args: {
    mode: LOGO_MODES.ICON,
    size: LOGO_SIZES.DISPLAY,
  },
};

export const Horizontal: Story = {
  args: {
    mode: LOGO_MODES.HORIZONTAL,
    size: LOGO_SIZES.DISPLAY,
  },
};
