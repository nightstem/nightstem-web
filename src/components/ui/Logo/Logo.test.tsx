import { axe } from 'vitest-axe';
import { render } from '@testing-library/react';

import { Logo } from '@/components/ui/Logo';
import type {
  LogoAnimation,
  LogoMode,
  LogoSize,
} from '@/components/ui/Logo/types';
import {
  LOGO_ANIMATIONS,
  LOGO_MODES,
  LOGO_SIZES,
} from '@/components/ui/Logo/constants';

const modes = Object.values(LOGO_MODES) as LogoMode[];
const animations = Object.values(LOGO_ANIMATIONS) as LogoAnimation[];

describe('Accessibility', () => {
  it.each(modes)(
    'should not have any accessibility violations in "%s" mode',
    async (mode) => {
      const { container } = render(<Logo mode={mode} />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    },
  );
});

describe('Snapshots', () => {
  const sizes = Object.values(LOGO_SIZES) as LogoSize[];

  it.each(modes.flatMap((mode) => sizes.map((size) => [mode, size] as const)))(
    'matches snapshot for %s mode with %s size',
    (mode, size) => {
      const { container } = render(<Logo mode={mode} size={size} />);
      expect(container.firstChild).toMatchSnapshot();
    },
  );

  it.each(animations)(
    'matches snapshot for HORIZONTAL mode with DISPLAY size and %s animation',
    (animation) => {
      const { container } = render(
        <Logo
          mode={LOGO_MODES.HORIZONTAL}
          size={LOGO_SIZES.DISPLAY}
          animation={animation}
        />,
      );
      expect(container.firstChild).toMatchSnapshot();
    },
  );
});
