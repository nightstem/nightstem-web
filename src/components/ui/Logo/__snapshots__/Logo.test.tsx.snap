// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshots > matches snapshot for FULL mode with DISPLAY size and none animation 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[4.488rem] w-[4.488rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-display"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for FULL mode with DISPLAY size and pulse animation 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[4.488rem] w-[4.488rem] animate-pulse shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-display"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with body size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[0.8rem] w-[0.8rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="text-body"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with caption size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[0.6rem] w-[0.6rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="text-caption"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with display size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[4.488rem] w-[4.488rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-display"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with hint size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[0.45rem] w-[0.45rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="text-hint"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with lg size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[1.895rem] w-[1.895rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-lg"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with md size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[1.422rem] w-[1.422rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-md"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with sm size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[1.066rem] w-[1.066rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-sm"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with xl size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[2.526rem] w-[2.526rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-xl"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for full mode with xxl size 1`] = `
<div
  class="inline-flex flex-col text-center items-end"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[3.367rem] w-[3.367rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-xxl"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with body size 1`] = `
<div
  class="inline-flex items-center gap-2"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[0.8rem] w-[0.8rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="text-body"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with caption size 1`] = `
<div
  class="inline-flex items-center gap-1"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[0.6rem] w-[0.6rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="text-caption"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with display size 1`] = `
<div
  class="inline-flex items-center gap-4"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[4.488rem] w-[4.488rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-display"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with hint size 1`] = `
<div
  class="inline-flex items-center gap-1"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[0.45rem] w-[0.45rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="text-hint"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with lg size 1`] = `
<div
  class="inline-flex items-center gap-3"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[1.895rem] w-[1.895rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-lg"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with md size 1`] = `
<div
  class="inline-flex items-center gap-2"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[1.422rem] w-[1.422rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-md"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with sm size 1`] = `
<div
  class="inline-flex items-center gap-2"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[1.066rem] w-[1.066rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-sm"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with xl size 1`] = `
<div
  class="inline-flex items-center gap-3"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[2.526rem] w-[2.526rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-xl"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for horizontal mode with xxl size 1`] = `
<div
  class="inline-flex items-center gap-3"
>
  <svg
    aria-hidden="true"
    class="text-primary-500 h-[3.367rem] w-[3.367rem] shrink-0"
    fill="none"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
  <span
    class="heading-xxl"
  >
    Nightstem
  </span>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with body size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[0.8rem] w-[0.8rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with caption size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[0.6rem] w-[0.6rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with display size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[4.488rem] w-[4.488rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with hint size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[0.45rem] w-[0.45rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with lg size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[1.895rem] w-[1.895rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with md size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[1.422rem] w-[1.422rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with sm size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[1.066rem] w-[1.066rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with xl size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[2.526rem] w-[2.526rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;

exports[`Snapshots > matches snapshot for icon mode with xxl size 1`] = `
<div
  class="inline-flex"
>
  <svg
    aria-label="Nightstem logo"
    class="text-primary-500 h-[3.367rem] w-[3.367rem] shrink-0"
    fill="none"
    role="img"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="currentColor"
      height="192.283"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 154.761 0)"
      width="219.719"
    />
    <rect
      fill="currentColor"
      height="428.961"
      transform="matrix(0.814781 0.579768 -0.80486 0.593464 346.04 135.875)"
      width="188.959"
    />
  </svg>
</div>
`;
